speech/speech_samples.rb

```diff
@@ -12,160 +12,168 @@
 # [START speech_sync_recognize_gcs]
-  # project_id   = "Your Google Cloud project ID"
   # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio storage_path, encoding:    :linear16,
-                                      sample_rate: 16000,
-                                      language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  results = audio.recognize
+  config = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio  = { uri: storage_path }
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  response = speech.recognize config, audio
+
+  alternatives = response.results.first.alternatives
+
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
   end
 # [END speech_sync_recognize_gcs]
 end
 
-def speech_async_recognize project_id: nil, audio_file_path: nil
+def speech_async_recognize audio_file_path: nil
 # [START speech_async_recognize]
-  # project_id      = "Your Google Cloud project ID"
   # audio_file_path = "Path to file on which to perform speech recognition"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio audio_file_path, encoding:    :linear16,
-                                         sample_rate: 16000,
-                                         language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  operation = audio.process
+  audio_file = File.binread audio_file_path
+  config     = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio      = { content: audio_file }
 
-  puts "Operation started"
+  operation = speech.long_running_recognize config, audio
 
   operation.wait_until_done!
 
-  results = operation.results
+  raise operation.results.message if operation.error?
 
-  results.each do |result|
-    puts "Transcription: #{result.transcript}"
+  alternatives = operation.results.first.alternatives
+
+  alternatives.each do |alternative|
+    puts "Transcription: #{alternative.transcript}"
   end
 # [END speech_async_recognize]
 end
 
-def speech_async_recognize_gcs project_id: nil, storage_path: nil
+def speech_async_recognize_gcs storage_path: nil
 # [START speech_async_recognize_gcs]
-  # project_id   = "Your Google Cloud project ID"
   # storage_path = "Path to file in Cloud Storage, eg. gs://bucket/audio.raw"
 
   require "google/cloud/speech"
 
-  speech = Google::Cloud::Speech.new project: project_id
-  audio  = speech.audio storage_path, encoding:    :linear16,
-                                      sample_rate: 16000,
-                                      language:    "en-US"
+  speech = Google::Cloud::Speech.new
 
-  operation = audio.process
+  config = { encoding: :LINEAR16, sample_rate_hertz: 16000, language_code: "en-US" }
+  audio  = { uri: storage_path }
```

Comment:
# ⚠️  Video uses `input_uri` and `input_content` but Speech uses `content` or `uri`